proteus/mprans/SpatialTools.py

```diff
@@ -471,161 +473,159 @@ def setDimensions(self, dim):
+        self.regionIndice['y+'] = r_i
+        regionFlags += [r_i+1]
         if y_p:
-            vertices += [[x0, y1+y_p, z0], [x1, y1+y_p, z0]]
-            segments += [[3, v_i], [v_i, v_i+1], [v_i+1, 2]]
-            facets += [[[3, 2, v_i+1, v_i]]]
-            regions += [[(x0+x1)/2., (y1+(y1+y_p))/2., (z0+z1)/2.]]
-            self.regionIndice['y+'] = r_i
-            regionFlags += [r_i+1]
-            v_i += 2
-            r_i += 1
-            nb_sponge += 1
+            facets += [[[2, 3, v_i+1, v_i]],
+                       [[6, 7, v_i+3, v_i+2]],
+                       [[2, 3, 7, 6]],
+                       [[2, v_i, v_i+2, 6]],
+                       [[3, v_i+1, v_i+3, 7]]]
+            facetFlags += [bt['z-'],
+                           bt['z+'],
+                           bt['sponge']]
+            if x_p > 0:
+                facetFlags += [bt['wall']]
+            else:
+                facetFlags += [bt['x+']]
+            if x_n > 0:
+                facetFlags += [bt['wall']]
+            else:
+                facetFlags += [bt['x-']]
+            volumes[0][0][-1] = f_i+3 
+            volumes += [[[f_i+i for i in range(6)]]]
+        v_i += 4
+        r_i += 1
+        nb_sponge += 1
+        f_i = len(facets)
+        # x+
+        vertices += [[x1+x_p, y0, z0], [x1+x_p, y1, z0],
+                     [x1+x_p, y0, z1], [x1+x_p, y1, z1]]
+        vertexFlags += [bt['z-'], bt['z-'],
+                        bt['z+'], bt['z+']]
+        volumes[0][0] += [f_i]  # add to volume
+        facets += [[[v_i, v_i+1, v_i+3, v_i+2]]]
+        facetFlags += [bt['x+']]
+        regions += [[(x1+(x1+x_p))/2., (y0+y1)/2., (z0+z1)/2.]]
+        self.regionIndice['x+'] = r_i
+        regionFlags += [r_i+1]
         if x_p:
-            vertices += [[x1+x_p, y0, z0], [x1+x_p, y1, z0]]
-            segments += [[1, v_i], [v_i, v_i+1], [v_i+1, 2]]
-            facets += [[[1, 2, v_i+1, v_i]]]
-            regions += [[(x1+(x1+x_p))/2., (y0+y1)/2., (z0+z1)/2.]]
-            self.regionIndice['x+'] = r_i
-            regionFlags += [r_i+1]
-            v_i += 2
-            r_i += 1
-            nb_sponge += 1
+            facets += [[[1, 2, v_i+1, v_i]],
+                       [[5, 6, v_i+3, v_i+2]],
+                       [[1, 2, 6, 5]],
+                       [[1, v_i, v_i+2, 5]],
+                       [[2, v_i+1, v_i+3, 6]]]
+            facetFlags += [bt['z-'],
+                           bt['z+'],
+                           bt['sponge']]
+            if y_n > 0:
+                facetFlags += [bt['wall']]
+            else:
+                facetFlags += [bt['y-']]
+            if y_p > 0:
+                facetFlags += [bt['wall']]
+            else:
+                facetFlags += [bt['y+']]
+            volumes[0][0][-1] = f_i+3 
+            volumes += [[[f_i+i for i in range(6)]]]
+        v_i += 4
+        r_i += 1
+        nb_sponge += 1
+        f_i = len(facets)
+        # x-
```

Comment:
Don't you need also to generate z- and z+ cubes?

Reply:
It is assumed that the gravity acceleration is aligned with the Z axis, like for most simulations, otherwise it adds an extra layer of complications

Reply:
- Having the gravity in the z direction only can be restrictive as some drawing will have the vertical in x or y.
- But I understand the logic. In this case you can put an error message if gravity is detected in x or y (non zero values). This will prevent users from doing something stupid

Reply:
I think it makes sense to leave it like this, and if someone wants to use it in another direction, they can rotate the shape by using `my_tank.rotate(angle, axis)` that is available in SpatialTools for any shape.

Reply:
Yes but if they do not realise that they need to align the gravity with z and e.g. give g=[9.81,0,0] will it still work?

Reply:
I mean that they can rotate the tank to be aligned with the gravity, it can be added to the docstring as a warning, but I think it is find as it is, as the `rotate` function should be quite straightforward to us

Reply:
I guess it should be fine for now, but let's keep an eye on the feedback we get from other users